superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx

```diff
@@ -90,13 +91,11 @@ export default function AddDataset() {
       })
     : undefined;
 
-  const getDatasetsList = () => {
-    SupersetClient.get({
-      endpoint: `/api/v1/dataset/?q=${queryParams}`,
-    })
-      .then(({ json }) => {
-        json.result.map((dataset: any) =>
-          setLinkedDatasets(linkedDatasets => [...linkedDatasets, dataset]),
+  const getDatasetsList = async () => {
+    await UseGetDatasetsList(queryParams)
+      .then(json => {
+        json?.result.map((dataset: DatasetObject) =>
```

Comment:
Bump on this comment before. Can you show the other method that you tried that didn’t work?